src/include/index/skiplist.h

```diff
@@ -372,9 +562,115 @@ class SkipList {
     } valueOrRoot_;
   };
 
-  // this one provides the abstract interfaces
-  class SkipListIterator {};
-  class ForwardIterator {};
+  /*
+   * class ForwardIterator - Iterator that supports forward iteration of the
+   * skip list
+   */
+  class ForwardIterator {
+   private:
+    SkipListInnerNode *node_;
```

Comment:
If you have any suggestion on the elements in this class, please tell me, thanks!